code/modules/modular_computers/file_system/programs/ntnrc_client.dm

```diff
@@ -249,9 +256,9 @@
 			data["strong"] = channel.strong
 			data["clients"] = clients
 			var/list/messages = list()
-			for(var/message in channel.messages)
+			for(var/i=channel.messages.len, i>1, i--)
```

Comment:
This should be `step -1` loop, also this will never get to `channel.messages[1]`. Is that intentional?